src/Amadevus.RecordGenerator.Generators/RecordGenerator.cs

```diff
@@ -4,46 +4,66 @@
+            ImmutableArray.Create(RecordPartialGenerator.Instance,
+                                  BuilderPartialGenerator.Instance,
+                                  DeconstructPartialGenerator.Instance);
+
         public RecordGenerator(AttributeData attributeData)
         {
             this.attributeData = attributeData;
         }
 
         public Task<SyntaxList<MemberDeclarationSyntax>> GenerateAsync(TransformationContext context, IProgress<Diagnostic> progress, CancellationToken cancellationToken)
         {
-            var generatedMembers = SyntaxFactory.List<MemberDeclarationSyntax>();
-            var features = GetFeatures();
-
-            if (context.ProcessingNode is ClassDeclarationSyntax classDeclaration)
-            {
-                var descriptor = classDeclaration.ToRecordDescriptor(features);
-                generatedMembers = generatedMembers
-                    .AddRange(
-                        GenerateRecordPartials(descriptor)
-                        .Where(x => x != null));
-            }
-            return Task.FromResult(generatedMembers);
-
-            IEnumerable<MemberDeclarationSyntax> GenerateRecordPartials(RecordDescriptor descriptor)
+            switch (context.ProcessingNode)
             {
-                if (descriptor.Entries.IsEmpty)
+                case ClassDeclarationSyntax classDeclaration:
                 {
-                    yield break;
+                    var descriptor = classDeclaration.ToRecordDescriptor();
+                    if (descriptor.Entries.IsEmpty)
+                        goto default;
+
+                    var generatedMembers = new List<MemberDeclarationSyntax>();
+                    var features = GetFeatures();
+                    var partialKeyword = Token(SyntaxKind.PartialKeyword);
+
+                    var partials =
+                        from g in PartialGenerators
+                        select g.Generate(descriptor, features) into g
+                        where g != null
+                        select
+                            ClassDeclaration(classDeclaration.Identifier.WithoutTrivia())
+                                .WithTypeParameterList(classDeclaration.TypeParameterList?.WithoutTrivia())
+                                .WithBaseList(g.BaseTypes.IsEmpty ? null : BaseList(SeparatedList(g.BaseTypes)))
+                                .WithModifiers(TokenList(g.Modifiers.Except(new[] { partialKeyword })
+                                                                    .Append(partialKeyword)))
+                                .WithMembers(List(g.Members))
+                                .AddGeneratedCodeAttributeOnMembers();
+
+                    foreach (var partial in partials)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        generatedMembers.Add(partial);
+                    }
+
+                    return Task.FromResult(List(generatedMembers));
```

Comment:
It's a very long `case`. I'd very much prefer it to be extracted to a local function.

Reply:
Okay. See 04a3efef48878e1c7d753460afc8be7bc3c7e9dc.